lib/parser.js

```diff
@@ -280,7 +298,7 @@ Parser.prototype._trimNodes = function (node) {
 
 Parser.prototype.includeFile = function (file, config) {
   let context = {};
-  context.cwf = file; // current working file
+  context.cwf = (config && config.targetFilePath) || file; // current working file
```

Comment:
If we are prioritising `targetFilePath` (which always has value) and so never using `file` ([which is `actualFilePath`](https://github.com/MarkBind/markbind-cli/pull/20/commits/a05647fc5842ed5dd3491c803d6adaca576a0cf3)), then why are we passing `actualFilePath` around?

Reply:
There is no config for https://github.com/MarkBind/markbind-cli/blob/master/lib/Page.js#L67. It uses `file`.

I was thinking if it is possible to not change markbind cli code and keep `this.dynamicIncludeSrc.push({from: context.cwf, to: filePath`. Then check for the boilerplate file only in `Parser.prototype.includeFile`. It doesn't seem to work for now

Reply:
You modified markbind-cli to pass `file` as `config.targetFilePath` and `actualFilePath` as `file`, then we use `config.targetFilePath` (i.e. `file`) so we're never using `file` (i.e. `actualFilePath`).

Then, why are we passing `actualFilePath` to this function as `file`?

Reply:
The `targetFilePath` is the local folder where the file is supposed to be. It is `file = source.target` in `resolveDependency` in markbind cli and is used to determine the `tempPath` and `resultPath` to save. It is also passed in `config.targetFilePath` so that the context of the children of dynamically included boilerplates files can be set to the local folder too.

The `actualFilePath` is where the file exists - it can be in the boilerplate folder. It is read in  `includeFile` in `fs.readFile(file, (err, data)`.  If we use `targetFilePath` and try to read from it in (unmodified) `includeFile` , it will throw an error.

Reply:
Thanks for clarifying. Let's rename `file` to `targetFilePath` in `Page.prototype.resolveDependency`:

```diff
- const file = dependency.target;
+ const targetFilePath = dependency.target;
  const actualFilePath = dependency.actualFilePath;

  return markbinder.includeFile(actualFilePath, {
    baseUrlMap: this.baseUrlMap,
    rootPath: this.rootPath,
-   targetFilePath: file,
+   targetFilePath: targetFilePath
  })
```

Reply:
> ```diff
> - const file = dependency.target;
> + const targetFilePath = dependency.target;
> const actualFilePath = dependency.actualFilePath;
>
>  return markbinder.includeFile(actualFilePath, {
>    baseUrlMap: this.baseUrlMap,
>    rootPath: this.rootPath,
> -   targetFilePath: file,
> +   targetFilePath: targetFilePath
>  })
> ```

can actually do

```diff
 return markbinder.includeFile(actualFilePath, {
   baseUrlMap: this.baseUrlMap,
    rootPath: this.rootPath,
-   targetFilePath: file,
+   targetFilePath
  })
```